src/main/kotlin/com/ajax/cryptocurrency/bpp/SingleShotBackgroundJobBeanPostProcessor.kt

```diff
@@ -0,0 +1,50 @@
+package com.ajax.cryptocurrency.bpp
+
+import com.ajax.cryptocurrency.annotation.SingleShotBackgroundJob
+import com.ajax.cryptocurrency.repository.CryptocurrencyRepository
+import com.ajax.cryptocurrency.parser.PriceSaver
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+@Component
+class SingleShotBackgroundJobBeanPostProcessor(
+    private val cryptocurrencyRepository: CryptocurrencyRepository,
+    @Value("\${cryptocurrency.name}") val cryptocurrencyNames: List<String>,
+    @Value("\${cryptocurrency.sleepTime}") private val sleepTime: String
+) : BeanPostProcessor {
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any {
+        return bean
+    }
+
+    override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
+        val annotation = bean::class.java.getAnnotation(SingleShotBackgroundJob::class.java)
```

Comment:
Borysov would be ashamed of you

Reply:
true, true

Reply:
<img width="858" alt="image" src="https://github.com/Wectro20/KotlinPetProject/assets/33807305/96f55b31-4f4f-477b-868c-9ef5ab19db97">

Reply:
was changed a long time ago
<img width="656" alt="image" src="https://github.com/Wectro20/KotlinPetProject/assets/33807305/688e821d-9229-465f-a2ad-96cec5d9fdeb">
some requested changes 9 upper